src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs

```diff
@@ -2014,7 +2014,55 @@ protected override void BeginProcessing()
             string targetMessage = StringUtil.Format(ProcessResources.StartProcessTarget, startInfo.FileName, startInfo.Arguments.Trim());
             if (!ShouldProcess(targetMessage)) { return; }
 
-            Process process = Start(startInfo);
+            Process process = null;
+
+#if !UNIX
+            ProcessCollection jobObject = null;
+            bool? jobAssigned = null;
+#endif
+            if (startInfo.UseShellExecute)
+            {
+                process = StartWithShellExecute(startInfo);
+            }
+            else
+            {
+#if UNIX
+                process = new Process() { StartInfo = startInfo };
+                SetupInputOutputRedirection(process);
+                process.Start();
```

Comment:
Do you know anything stopping us from direct using process.Start() on Windows too?

I see only limitation - WindowStyle. .Net API support this only for shellexecution. We could ask .Net team to enhance the API and then we could remove the pinvoke at all.

Reply:
It should be technically possible. I originally assumed the biggest blocker was that the dotnet API didn't support specifying the redirection file paths and the code would have to manually do all that. But it seems like it already does that for the non-Windows side so we could just reuse that. There could be other things that might crop up but I preferred to just address the original problem in question and potentially look at the larger change in the future.

Reply:
Thanks for confirmation! I will open new issue in .Net repo. If they confirm we can continue in more simple way.

Reply:
Isuue https://github.com/dotnet/runtime/issues/81681